src/openms/source/FORMAT/SqliteConnector.cpp

```diff
@@ -131,26 +128,28 @@ namespace OpenMS
   {
     int rc;
     sqlite3_stmt *stmt = nullptr;
-    SqliteConnector::executePreparedStatement(db, &stmt, prepare_statement);
+    prepareStatement(db, &stmt, prepare_statement);
     for (Size k = 0; k < data.size(); k++)
     {
       // Fifth argument is a destructor for the blob.
       // SQLITE_STATIC because the statement is finalized
       // before the buffer is freed:
-      rc = sqlite3_bind_blob(stmt, k+1, data[k].c_str(), data[k].size(), SQLITE_STATIC);
+      rc = sqlite3_bind_blob(stmt, k+1, data[k].c_str(), (int)data[k].size(), SQLITE_STATIC);
       if (rc != SQLITE_OK)
       {
         std::cerr << "SQL error after sqlite3_bind_blob at iteration " << k << std::endl;
         std::cerr << "Prepared statement " << prepare_statement << std::endl;
+        // TODO this is a mem-leak (missing sqlite3_finalize())
         throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, sqlite3_errmsg(db));
-      } 
+      }
     }
 
     rc = sqlite3_step(stmt);
     if (rc != SQLITE_DONE)
     {
       std::cerr << "SQL error after sqlite3_step" << std::endl;
       std::cerr << "Prepared statement " << prepare_statement << std::endl;
+      // TODO this is a mem-leak (missing sqlite3_finalize())
```

Comment:
Why not call it?

Reply:
because that only fixes the problem locally. We should think about a more thorough solution?!

Reply:
Something like unique_ptr + lambda deleter?

Reply:
yes. baked into `prepareStatement()` or some other place appropriate...

Reply:
If you are here and throw an exception in the SQL code (e.g. invalid SQL), you probably have much bigger problems than a memory leak?